Build error on macOS 10.15.5 Xcode 11.5

Steps taken:

```
% git clone --recursive git://github.com/cinder/Cinder.git cinder_master
% cd cinder_master/proj/xcode
% ./fullbuild.sh
```

Lots of normal build output that ends with:

```
Libtool /Users/howes/src/cinder_master/lib/macosx/Debug/libcinder.a normal x86_64 (in target 'cinder' from project 'cinder')
    cd /Users/howes/src/cinder_master/proj/xcode
    export MACOSX_DEPLOYMENT_TARGET=10.8
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool -static -arch_only x86_64 -D -syslibroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk -L/Users/howes/src/cinder_master/lib/macosx/Debug -filelist /Users/howes/src/cinder_master/proj/xcode/build/cinder.build/Debug/cinder.build/Objects-normal/x86_64/cinder.LinkFileList -framework AppKit -framework Cocoa -framework OpenGL -framework QTKit -framework CoreVideo -framework CoreGraphics -framework ApplicationServices -framework ScreenSaver -framework AudioToolbox -framework AudioUnit -framework CoreAudio -dependency_info /Users/howes/src/cinder_master/proj/xcode/build/cinder.build/Debug/cinder.build/Objects-normal/x86_64/cinder_libtool_dependency_info.dat -o /Users/howes/src/cinder_master/lib/macosx/Debug/libcinder.a

** BUILD SUCCEEDED **

Command line invocation:
    /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -project /Users/howes/src/cinder_master/proj/xcode/cinder.xcodeproj -target cinder_iphone -configuration Release -sdk iphoneos

Build settings from command line:
    SDKROOT = iphoneos13.5

note: Using new build system
note: Planning build
note: Constructing build description
error: AppKit is not available when building for iOS. (in target 'cinder_iphone' from project 'cinder')
error: Cocoa is not available when building for iOS. (in target 'cinder_iphone' from project 'cinder')

** BUILD FAILED **

Command line invocation:
    /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -project /Users/howes/src/cinder_master/proj/xcode/cinder.xcodeproj -target cinder_iphone -configuration Debug -sdk iphoneos

Build settings from command line:
    SDKROOT = iphoneos13.5

note: Using new build system
note: Planning build
note: Constructing build description
error: AppKit is not available when building for iOS. (in target 'cinder_iphone' from project 'cinder')
error: Cocoa is not available when building for iOS. (in target 'cinder_iphone' from project 'cinder')

** BUILD FAILED **

Command line invocation:
    /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -project /Users/howes/src/cinder_master/proj/xcode/cinder.xcodeproj -target cinder_iphone_sim -configuration Release -sdk iphonesimulator

Build settings from command line:
    SDKROOT = iphonesimulator13.5

note: Using new build system
note: Planning build
note: Constructing build description
warning: Product libcinder.a cannot weak-link framework CoreVideo.framework (in target 'cinder_iphone_sim' from project 'cinder')
warning: Product libcinder.a cannot weak-link framework AVFoundation.framework (in target 'cinder_iphone_sim' from project 'cinder')
warning: Product libcinder.a cannot weak-link framework CoreMedia.framework (in target 'cinder_iphone_sim' from project 'cinder')
warning: Product libcinder.a cannot weak-link framework CoreVideo.framework (in target 'cinder_iphone_sim' from project 'cinder')
warning: Product libcinder.a cannot weak-link framework AVFoundation.framework (in target 'cinder_iphone_sim' from project 'cinder')
warning: Product libcinder.a cannot weak-link framework CoreMedia.framework (in target 'cinder_iphone_sim' from project 'cinder')
error: AppKit is not available when building for iOS Simulator. (in target 'cinder_iphone_sim' from project 'cinder')
error: Cocoa is not available when building for iOS Simulator. (in target 'cinder_iphone_sim' from project 'cinder')
error: QTKit is not available when building for iOS Simulator. (in target 'cinder_iphone_sim' from project 'cinder')

** BUILD FAILED **

Command line invocation:
    /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -project /Users/howes/src/cinder_master/proj/xcode/cinder.xcodeproj -target cinder_iphone_sim -configuration Debug -sdk iphonesimulator

Build settings from command line:
    SDKROOT = iphonesimulator13.5

note: Using new build system
note: Planning build
note: Constructing build description
warning: Product libcinder.a cannot weak-link framework CoreVideo.framework (in target 'cinder_iphone_sim' from project 'cinder')
warning: Product libcinder.a cannot weak-link framework AVFoundation.framework (in target 'cinder_iphone_sim' from project 'cinder')
warning: Product libcinder.a cannot weak-link framework CoreMedia.framework (in target 'cinder_iphone_sim' from project 'cinder')
warning: Product libcinder.a cannot weak-link framework CoreVideo.framework (in target 'cinder_iphone_sim' from project 'cinder')
warning: Product libcinder.a cannot weak-link framework AVFoundation.framework (in target 'cinder_iphone_sim' from project 'cinder')
warning: Product libcinder.a cannot weak-link framework CoreMedia.framework (in target 'cinder_iphone_sim' from project 'cinder')
error: AppKit is not available when building for iOS Simulator. (in target 'cinder_iphone_sim' from project 'cinder')
error: Cocoa is not available when building for iOS Simulator. (in target 'cinder_iphone_sim' from project 'cinder')
error: QTKit is not available when building for iOS Simulator. (in target 'cinder_iphone_sim' from project 'cinder')

** BUILD FAILED **
```

--- a/include/cinder/app/cocoa/PlatformCocoa.h
+++ b/include/cinder/app/cocoa/PlatformCocoa.h
@@ -123,7 +123,6 @@ class PlatformCocoa : public Platform {
 #endif
 
   private:
-	NSAutoreleasePool*		mAutoReleasePool;
 	mutable NSBundle*		mBundle;
 	
 	bool					mDisplaysInitialized;

--- a/proj/xcode/cinder.xcodeproj/project.pbxproj
+++ b/proj/xcode/cinder.xcodeproj/project.pbxproj
@@ -621,9 +621,6 @@
 		27C100C21BD16D4800AF387F /* CoreVideo.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 0074399D0EA7BB7D005DD3E6 /* CoreVideo.framework */; settings = {ATTRIBUTES = (Weak, ); }; };
 		27C100C31BD16D4800AF387F /* AVFoundation.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = C7B9305B121C94E900093AFE /* AVFoundation.framework */; settings = {ATTRIBUTES = (Weak, ); }; };
 		27C100C41BD16D4800AF387F /* CoreMedia.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = C7B92EF5121C828400093AFE /* CoreMedia.framework */; settings = {ATTRIBUTES = (Weak, ); }; };
-		27C100C51BD16D4800AF387F /* AppKit.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 0867D6A5FE840307C02AAC07 /* AppKit.framework */; };
-		27C100C61BD16D4800AF387F /* Cocoa.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 1058C7B1FEA5585E11CA2CBB /* Cocoa.framework */; };
-		27C100C71BD16D4800AF387F /* QTKit.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 007439920EA7BB47005DD3E6 /* QTKit.framework */; };
 		27C100C81BD16D4800AF387F /* CoreGraphics.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 00E0B4B10F605D64002C8FBD /* CoreGraphics.framework */; };
 		27C100C91BD16D4800AF387F /* AudioToolbox.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = C7BA40880FA8C13A00AADC07 /* AudioToolbox.framework */; };
 		27C100CA1BD16D4800AF387F /* AudioUnit.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = C7BA408A0FA8C13A00AADC07 /* AudioUnit.framework */; };
@@ -954,8 +951,6 @@
 		27C1FF681BD0AE3400AF387F /* ImageSourceFileQuartz.cpp in Sources */ = {isa = PBXBuildFile; fileRef = 009FD55610CAB8B700D63B1B /* ImageSourceFileQuartz.cpp */; };
 		27C1FF6A1BD0AE3400AF387F /* vorbisfile.c in Sources */ = {isa = PBXBuildFile; fileRef = 111A5E95191F703D005C3166 /* vorbisfile.c */; settings = {COMPILER_FLAGS = "-Wno-unused -Wno-conversion"; }; };
 		27C1FF6C1BD0AE3400AF387F /* CoreMedia.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = C7B92EF5121C828400093AFE /* CoreMedia.framework */; };
-		27C1FF6D1BD0AE3400AF387F /* AppKit.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 0867D6A5FE840307C02AAC07 /* AppKit.framework */; };
-		27C1FF6E1BD0AE3400AF387F /* Cocoa.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 1058C7B1FEA5585E11CA2CBB /* Cocoa.framework */; };
 		27C1FF6F1BD0AE3400AF387F /* CoreGraphics.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 00E0B4B10F605D64002C8FBD /* CoreGraphics.framework */; };
 		27C1FF701BD0AE3400AF387F /* AudioToolbox.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = C7BA40880FA8C13A00AADC07 /* AudioToolbox.framework */; };
 		27C1FF711BD0AE3400AF387F /* AudioUnit.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = C7BA408A0FA8C13A00AADC07 /* AudioUnit.framework */; };
@@ -2352,9 +2347,6 @@
 				27C100C21BD16D4800AF387F /* CoreVideo.framework in Frameworks */,
 				27C100C31BD16D4800AF387F /* AVFoundation.framework in Frameworks */,
 				27C100C41BD16D4800AF387F /* CoreMedia.framework in Frameworks */,
-				27C100C51BD16D4800AF387F /* AppKit.framework in Frameworks */,
-				27C100C61BD16D4800AF387F /* Cocoa.framework in Frameworks */,
-				27C100C71BD16D4800AF387F /* QTKit.framework in Frameworks */,
 				27C100C81BD16D4800AF387F /* CoreGraphics.framework in Frameworks */,
 				27C100C91BD16D4800AF387F /* AudioToolbox.framework in Frameworks */,
 				27C100CA1BD16D4800AF387F /* AudioUnit.framework in Frameworks */,
@@ -2367,8 +2359,6 @@
 			buildActionMask = 2147483647;
 			files = (
 				27C1FF6C1BD0AE3400AF387F /* CoreMedia.framework in Frameworks */,
-				27C1FF6D1BD0AE3400AF387F /* AppKit.framework in Frameworks */,
-				27C1FF6E1BD0AE3400AF387F /* Cocoa.framework in Frameworks */,
 				27C1FF6F1BD0AE3400AF387F /* CoreGraphics.framework in Frameworks */,
 				27C1FF701BD0AE3400AF387F /* AudioToolbox.framework in Frameworks */,
 				27C1FF711BD0AE3400AF387F /* AudioUnit.framework in Frameworks */,
@@ -4756,7 +4746,7 @@
 		0867D690FE84028FC02AAC07 /* Project object */ = {
 			isa = PBXProject;
 			attributes = {
-				LastUpgradeCheck = 1130;
+				LastUpgradeCheck = 1150;
 			};
 			buildConfigurationList = 1DEB922208733DC00010E9CD /* Build configuration list for PBXProject "cinder" */;
 			compatibilityVersion = "Xcode 3.2";
@@ -5573,12 +5563,11 @@
 			isa = XCBuildConfiguration;
 			buildSettings = {
 				ALWAYS_SEARCH_USER_PATHS = NO;
-				ARCHS = "$(ARCHS_STANDARD)";
 				CHECKED_STL = "-D_GLIBCXX_DEBUG -D_GLIBCXX_DEBUG_PEDANTIC";
+				CLANG_ENABLE_OBJC_WEAK = YES;
 				COMBINE_HIDPI_IMAGES = YES;
 				COPY_PHASE_STRIP = NO;
 				GCC_DYNAMIC_NO_PIC = NO;
-				GCC_ENABLE_OBJC_GC = unsupported;
 				GCC_OPTIMIZATION_LEVEL = 0;
 				GCC_PRECOMPILE_PREFIX_HEADER = YES;
 				GCC_PREFIX_HEADER = cinder_Prefix.pch;
@@ -5612,13 +5601,12 @@
 			isa = XCBuildConfiguration;
 			buildSettings = {
 				ALWAYS_SEARCH_USER_PATHS = NO;
-				ARCHS = "$(ARCHS_STANDARD)";
+				CLANG_ENABLE_OBJC_WEAK = YES;
 				COMBINE_HIDPI_IMAGES = YES;
 				COPY_PHASE_STRIP = NO;
 				DEAD_CODE_STRIPPING = YES;
 				DEBUG_INFORMATION_FORMAT = "dwarf-with-dsym";
 				GCC_DEBUGGING_SYMBOLS = full;
-				GCC_ENABLE_OBJC_GC = unsupported;
 				GCC_FAST_MATH = YES;
 				GCC_OPTIMIZATION_LEVEL = 3;
 				GCC_PRECOMPILE_PREFIX_HEADER = YES;
@@ -5653,17 +5641,29 @@
 			isa = XCBuildConfiguration;
 			buildSettings = {
 				ALWAYS_SEARCH_USER_PATHS = NO;
+				CLANG_ANALYZER_LOCALIZABILITY_NONLOCALIZED = YES;
 				CLANG_CXX_LANGUAGE_STANDARD = "c++0x";
 				CLANG_CXX_LIBRARY = "libc++";
+				CLANG_WARN_BLOCK_CAPTURE_AUTORELEASING = YES;
 				CLANG_WARN_BOOL_CONVERSION = YES;
+				CLANG_WARN_COMMA = YES;
 				CLANG_WARN_CONSTANT_CONVERSION = YES;
+				CLANG_WARN_DEPRECATED_OBJC_IMPLEMENTATIONS = YES;
 				CLANG_WARN_EMPTY_BODY = YES;
 				CLANG_WARN_ENUM_CONVERSION = YES;
+				CLANG_WARN_INFINITE_RECURSION = YES;
 				CLANG_WARN_INT_CONVERSION = YES;
+				CLANG_WARN_NON_LITERAL_NULL_CONVERSION = YES;
+				CLANG_WARN_OBJC_IMPLICIT_RETAIN_SELF = YES;
+				CLANG_WARN_OBJC_LITERAL_CONVERSION = YES;
+				CLANG_WARN_RANGE_LOOP_ANALYSIS = YES;
+				CLANG_WARN_STRICT_PROTOTYPES = YES;
+				CLANG_WARN_SUSPICIOUS_MOVE = YES;
 				CLANG_WARN_UNREACHABLE_CODE = YES;
 				CLANG_WARN__DUPLICATE_METHOD_MATCH = YES;
 				CONFIGURATION_BUILD_DIR = "$(BUILD_DIR)";
 				ENABLE_STRICT_OBJC_MSGSEND = YES;
+				ENABLE_TESTABILITY = YES;
 				GCC_INLINES_ARE_PRIVATE_EXTERN = YES;
 				GCC_NO_COMMON_BLOCKS = YES;
 				GCC_OBJC_CALL_CXX_CDTORS = YES;
@@ -5683,6 +5683,7 @@
 					../../include/oggvorbis,
 				);
 				OBJROOT = build/;
+				ONLY_ACTIVE_ARCH = YES;
 				OTHER_CPLUSPLUSFLAGS = "$(OTHER_CFLAGS)";
 				USER_HEADER_SEARCH_PATHS = "../../include ../../src/oggvorbis/vorbis ../../src/r8brain";
 			};
@@ -5692,13 +5693,24 @@
 			isa = XCBuildConfiguration;
 			buildSettings = {
 				ALWAYS_SEARCH_USER_PATHS = NO;
+				CLANG_ANALYZER_LOCALIZABILITY_NONLOCALIZED = YES;
 				CLANG_CXX_LANGUAGE_STANDARD = "c++0x";
 				CLANG_CXX_LIBRARY = "libc++";
+				CLANG_WARN_BLOCK_CAPTURE_AUTORELEASING = YES;
 				CLANG_WARN_BOOL_CONVERSION = YES;
+				CLANG_WARN_COMMA = YES;
 				CLANG_WARN_CONSTANT_CONVERSION = YES;
+				CLANG_WARN_DEPRECATED_OBJC_IMPLEMENTATIONS = YES;
 				CLANG_WARN_EMPTY_BODY = YES;
 				CLANG_WARN_ENUM_CONVERSION = YES;
+				CLANG_WARN_INFINITE_RECURSION = YES;
 				CLANG_WARN_INT_CONVERSION = YES;
+				CLANG_WARN_NON_LITERAL_NULL_CONVERSION = YES;
+				CLANG_WARN_OBJC_IMPLICIT_RETAIN_SELF = YES;
+				CLANG_WARN_OBJC_LITERAL_CONVERSION = YES;
+				CLANG_WARN_RANGE_LOOP_ANALYSIS = YES;
+				CLANG_WARN_STRICT_PROTOTYPES = YES;
+				CLANG_WARN_SUSPICIOUS_MOVE = YES;
 				CLANG_WARN_UNREACHABLE_CODE = YES;
 				CLANG_WARN__DUPLICATE_METHOD_MATCH = YES;
 				CONFIGURATION_BUILD_DIR = "$(BUILD_DIR)";
@@ -5732,6 +5744,7 @@
 			buildSettings = {
 				ALTERNATE_PERMISSIONS_FILES = i386;
 				ALWAYS_SEARCH_USER_PATHS = NO;
+				CLANG_ENABLE_OBJC_WEAK = YES;
 				COPY_PHASE_STRIP = NO;
 				GCC_DYNAMIC_NO_PIC = NO;
 				GCC_OPTIMIZATION_LEVEL = 0;
@@ -5769,6 +5782,7 @@
 			buildSettings = {
 				ALTERNATE_PERMISSIONS_FILES = i386;
 				ALWAYS_SEARCH_USER_PATHS = NO;
+				CLANG_ENABLE_OBJC_WEAK = YES;
 				COPY_PHASE_STRIP = NO;
 				DEAD_CODE_STRIPPING = YES;
 				DEBUG_INFORMATION_FORMAT = "dwarf-with-dsym";
@@ -5808,6 +5822,7 @@
 			isa = XCBuildConfiguration;
 			buildSettings = {
 				ALWAYS_SEARCH_USER_PATHS = NO;
+				CLANG_ENABLE_OBJC_WEAK = YES;
 				COPY_PHASE_STRIP = NO;
 				ENABLE_BITCODE = NO;
 				GCC_DYNAMIC_NO_PIC = NO;
@@ -5846,6 +5861,7 @@
 			isa = XCBuildConfiguration;
 			buildSettings = {
 				ALWAYS_SEARCH_USER_PATHS = NO;
+				CLANG_ENABLE_OBJC_WEAK = YES;
 				COPY_PHASE_STRIP = NO;
 				DEAD_CODE_STRIPPING = YES;
 				DEBUG_INFORMATION_FORMAT = "dwarf-with-dsym";

--- a/src/AntTweakBar/TwMgr.cpp
+++ b/src/AntTweakBar/TwMgr.cpp
@@ -6399,8 +6399,6 @@ CTwMgr::CCursor CTwMgr::PixmapCursor(int _CurIdx)
     [img addRepresentation: imgr];
     NSCursor *cur = [[NSCursor alloc] initWithImage: img hotSpot: NSMakePoint(g_CurHot[_CurIdx][0],g_CurHot[_CurIdx][1])];
 
-    [imgr autorelease];
-    [img autorelease];
     if (cur)
         return cur;
     else

--- a/src/cinder/app/Renderer.cpp
+++ b/src/cinder/app/Renderer.cpp
@@ -72,15 +72,11 @@ Renderer2d::Renderer2d()
 #if defined( CINDER_MAC )
 Renderer2d::~Renderer2d()
 {
-	if( mImpl )
-		::CFRelease( mImpl );
 }
 
 void Renderer2d::setup( CGRect frame, NSView *cinderView, RendererRef /*sharedRenderer*/, bool retinaEnabled )
 {
 	mImpl = [[RendererImpl2dMacQuartz alloc] initWithFrame:NSRectFromCGRect(frame) cinderView:cinderView];
-	// This is necessary for Objective-C garbage collection to do the right thing
-	::CFRetain( mImpl );
 }
 
 #else

--- a/src/cinder/app/RendererGl.cpp
+++ b/src/cinder/app/RendererGl.cpp
@@ -67,16 +67,12 @@ RendererGl::RendererGl( const RendererGl &renderer )
 #if defined( CINDER_MAC )
 RendererGl::~RendererGl()
 {
-	if( mImpl )
-		::CFRelease( mImpl );
 }
 
 void RendererGl::setup( CGRect frame, NSView *cinderView, RendererRef sharedRenderer, bool retinaEnabled )
 {
 	RendererGlRef sharedGl = std::dynamic_pointer_cast<RendererGl>( sharedRenderer );
 	mImpl = [[RendererImplGlMac alloc] initWithFrame:NSRectFromCGRect(frame) cinderView:cinderView renderer:this sharedRenderer:sharedGl withRetina:retinaEnabled];
-	// This is necessary for Objective-C garbage collection to do the right thing
-	::CFRetain( mImpl );
 }
 
 void RendererGl::startDraw()

--- a/src/cinder/app/cocoa/PlatformCocoa.cpp
+++ b/src/cinder/app/cocoa/PlatformCocoa.cpp
@@ -52,8 +52,6 @@ PlatformCocoa::PlatformCocoa()
 	, mInsideModalLoop( false )
 #endif
 {
-	mAutoReleasePool = [[NSAutoreleasePool alloc] init];
-	
 	// This is necessary to force the linker not to strip these symbols from libboost_filesystem.a,
 	// which in turn would force users to explicitly link to that lib from their own apps.
 	auto dummy2 = fs::temp_directory_path();
@@ -72,7 +70,6 @@ void PlatformCocoa::prepareLaunch()
 
 void PlatformCocoa::cleanupLaunch()
 {
-	[mAutoReleasePool drain];
 }
 
 void PlatformCocoa::setBundle( NSBundle *bundle )

--- a/src/cinder/qtime/QuickTimeImplAvf.mm
+++ b/src/cinder/qtime/QuickTimeImplAvf.mm
@@ -699,12 +699,9 @@ void MovieBase::processAssetTracks( AVAsset* asset )
 
 void MovieBase::createPlayerItemOutput( const AVPlayerItem* playerItem )
 {
-	AVPlayerItemVideoOutput *oldPlayerVideoOutput = mPlayerVideoOutput;
 	mPlayerVideoOutput = [[AVPlayerItemVideoOutput alloc] initWithPixelBufferAttributes:avPlayerItemOutputDictionary()];
-	[oldPlayerVideoOutput release];
 	dispatch_queue_t outputQueue = dispatch_queue_create("movieVideoOutputQueue", DISPATCH_QUEUE_SERIAL);
 	[mPlayerVideoOutput setDelegate:mPlayerDelegate queue:outputQueue];
-	dispatch_release(outputQueue);
 	mPlayerVideoOutput.suppressesPlayerRendering = YES;
 	[playerItem addOutput:mPlayerVideoOutput];
 }

